File: 2018/Day1/Day1.py
input = []

with open("PuzzleInput.txt", "r") as f:
    for line in f:
        input.append(int(line))


def day_one_part_one(input):
    sum = 0

    for number in input:
        sum = sum + number

    print("The answer to part one is ", sum)


def day_one_part_two(input):
    part_one = 0
    part_two = 0
    frequency = []

    while part_two == 0:

        for number in input:
            part_one = part_one + number

            if part_one in frequency:
                part_two = part_one
                break
            else:
                frequency.append(part_one)

    print("The answer to part two is ", part_two)


if __name__ == "__main__":
    day_one_part_one(input)
    day_one_part_two(input)

File: 2021/Day1/Day1.java
import java.io.File;
import java.util.Scanner;
import java.util.ArrayList;

public class Day1{
  public static void main(String[] args) {
    Day1 day1 = new Day1();
    File puzzleInput = new File("PuzzleInput.txt");
    ArrayList<Integer> puzzleArray = day1.putInIntArrayList(puzzleInput);
    int partOne = day1.partOne(puzzleArray);
    int partTwo = day1.partTwo(puzzleArray);

    System.out.println("The answer to part 1 is " + partOne);
    System.out.println("The answer to part 2 is " + partTwo);
  }

  private int partOne(ArrayList<Integer> puzzleArray) {
    int increase = 0;

    for (int i = 1; i < puzzleArray.size(); i++) {
      if (puzzleArray.get(i) > puzzleArray.get(i - 1))
        increase++;
    }

    return increase;
  }

  private int partTwo(ArrayList<Integer> puzzleArray) {
    int increase = 0;
    int previousDepth = puzzleArray.get(0) + puzzleArray.get(1) + puzzleArray.get(2);

    for (int i = 1; i < puzzleArray.size()-2; i++) {
      int newDepth = puzzleArray.get(i) + puzzleArray.get(i + 1) + puzzleArray.get(i +2);
      if (newDepth > previousDepth) {
        increase++;
      }
      previousDepth = newDepth;
    }

    return increase;
  }

  private ArrayList<Integer> putInIntArrayList(File puzzleInput) {
    ArrayList<Integer> input = new ArrayList<Integer>();
    try {
      Scanner sc = new Scanner(puzzleInput);

      while (sc.hasNextLine()) {
        input.add(Integer.valueOf(sc.nextLine().replace("\n", "")));
      }
    } catch (Exception e) { }

    return input;
  }
}